core/application/bloodDonationWorkflow/DonorSearchService.ts

```diff
@@ -0,0 +1,181 @@
+import {
+  DonationStatus,
+  DonorSearchDTO
+} from '../../../commons/dto/DonationDTO'
+import Repository from '../models/policies/repositories/Repository'
+import { getGeohashNthNeighbors } from '../utils/geohash'
+import {
+  QueryConditionOperator,
+  QueryInput
+} from '../models/policies/repositories/QueryTypes'
+import {
+  DonorRoutingAttributes,
+  StepFunctionInput
+} from './Types'
+import { StepFunctionModel } from '../models/stepFunctions/StepFunctionModel'
+import { DONOR_SEARCH_PK_PREFIX } from '../models/dbModels/DonorSearchModel'
+import { AcceptedDonationFields } from '../models/dbModels/AcceptDonationModel'
+import { LocationDTO, UserDetailsDTO } from '../../../commons/dto/UserDTO'
+import { getBloodRequestMessage } from './BloodDonationMessages'
+import LocationModel from '../models/dbModels/LocationModel'
+
+export class DonorSearchService {
+  async routeDonorRequest(
+    donorRoutingAttributes: DonorRoutingAttributes,
+    sourceQueueArn: string,
+    userProfile: UserDetailsDTO,
+    donorSearchRepository: Repository<DonorSearchDTO>,
+    stepFunctionModel: StepFunctionModel
+  ): Promise<void> {
+    const { seekerId, requestPostId, createdAt } = donorRoutingAttributes
+
+    const donorSearchRecord = await donorSearchRepository.getItem(
+      `${DONOR_SEARCH_PK_PREFIX}#${seekerId}`,
+      `${DONOR_SEARCH_PK_PREFIX}#${createdAt}#${requestPostId}`
+    )
+
+    if (donorSearchRecord === null) {
+      await donorSearchRepository.create({
+        id: donorRoutingAttributes.requestPostId,
+        ...donorRoutingAttributes,
+        status: DonationStatus.PENDING,
+        retryCount: 0
+      })
+    }
+    const isDonationUpdateRequest = sourceQueueArn === process.env.DONOR_SEARCH_QUEUE_ARN
+    const hasDonationCompleted = donorSearchRecord !== null && donorSearchRecord.status === DonationStatus.COMPLETED
+    const bloodQuantityUpdated = donorSearchRecord !== null && donorRoutingAttributes.bloodQuantity > donorSearchRecord.bloodQuantity
```

Comment:
this is a little misleading, bloodQuantityUpdated? 
how would this happen?
someone initially wanted 5 bags now they want 6? that is a bit strange

Reply:
what we discussed that user can update the blood quantity. then we should run the donor search again